tests/datastructures/test_dsbsparse.py

```diff
@@ -173,6 +173,34 @@ def test_getitem(
         reference = dense[..., *accessed_element]
         assert xp.allclose(reference, dsbsparse[accessed_element])
 
+    @pytest.mark.usefixtures("factor")
+    def test_getitem_with_array(
+        self,
+        dsbsparse_type: DSBSparse,
+        block_sizes: xp.ndarray,
+        global_stack_shape: tuple,
+        factor: float,
+    ):
+        """Tests that we can get elements from a DSBSparse matrix using an tuple with arrays of row and column indices."""
+        coo = _create_coo(block_sizes)
+        dsbsparse = dsbsparse_type.from_sparray(
+            coo,
+            block_sizes=block_sizes,
+            global_stack_shape=global_stack_shape,
+        )
+
+        indices = xp.random.randint(
+            0, coo.shape[-1], size=(2, int(coo.shape[-1] * factor))
```

Comment:
This could generate repeated indices, right? So (if unlucky) it could be something like
```python
>>> indices
array([[ 5,  6,  5],
       [10,  3, 10]])
```
When i test this (on the kernels branch) it doesn't seem to give the correct result (i think we implicitly assume the indices are unique in the `find_inds` kernel). Example:
```python
>>> coo.toarray()[random_rows, random_cols]
array([0.87815981, 0.14031535, 0.87815981])
>>> dsbcoo[random_rows, random_cols][0]
array([0.        , 0.14031535, 0.87815981])  # :(
```

Should we support this? If yes, how? If no, could you ensure the test does not produce duplicate indices?

I think it would be good to support in principle, but i don't know how annoying it is to actually implement. I can have a look next week and try to fix it in #58

Reply:
Ah, I see. I only tested it on the dev branch, and there it is no issue if the same index is repeated. I didn't test it on the kernel branch. I can have a look on the kernel branch and see if I can figure out a nice solution for repeating indices.